docker/http-to-bucket/spec/app_spec.rb

```diff
@@ -316,4 +316,72 @@ def build_request_url(path = "/", params = {})
     end # context "and an unexpected error occurs"
 
   end # context "when all required parameters are provided"
-end # RSpec.describe
+
+  describe "#clean" do
+    context "when it's given a single layer hash" do
+      it "correctly removes empty hash elements" do
+        object = {foo: {}, bar: "baz"}
+        expectation = {bar: "baz"}
+
+        expect(clean(object)).to eq(expectation)
+      end
+    end
+
+    context "when it's given an array of hashes" do
+      it "correctly removes empty hash elements" do
+        object = [{foo: {}, bar: "baz"}, {foo: "bar", baz: {}}]
+        expectation = [{bar: "baz"}, {foo: "bar"}]
+
+        expect(clean(object)).to eq(expectation)
+      end
+    end
+
+    context "when it's given a multi layered hash" do
+      it "correctly removes empty hash elements" do
+        object = {
+          a: {foo: {}, bar: "baz"},
+          b: {foo: "bar", baz: {}},
+          c: {foo: {bar: "baz", foo: {}}}
+        }
+        expectation = {
+          a: {bar: "baz"},
+          b: {foo: "bar"},
+          c: {foo: {bar: "baz"}}
+        }
+
+        expect(clean(object)).to eq(expectation)
+      end
+    end
+
+    context "when it's given an array of many multi layered hash" do
+      it "correctly removes empty hash elements" do
+        object = [
+          {
+            a: {foo: {}, bar: "baz"},
+            b: {foo: "bar", baz: {}},
+            c: {foo: {bar: "baz", foo: {}}}
+          },
+          {
+            a: {foo: {}, bar: "baz"},
+            b: {foo: "bar", baz: {}},
+            c: {foo: {bar: "baz", foo: {}}}
+          },
+        ]
+        expectation = [
+          {
+          a: {bar: "baz"},
+          b: {foo: "bar"},
+          c: {foo: {bar: "baz"}}
+          },
+          {
+          a: {bar: "baz"},
+          b: {foo: "bar"},
+          c: {foo: {bar: "baz"}}
+          },
+        ]
+
+        expect(clean(object)).to eq(expectation)
+      end
+    end
+  end
```

Comment:
Please could you add a test that it doesn't remove empty arrays or empty strings?

```
object = {foo: []}
expectation = {foo: []}
```
```
object = {foo: ""}
expectation = {foo: ""}
```

Reply:
Another thought: it might be worth testing this:
```
object = {foo: {bar: {}}
expectation = {}
```

Reply:
Resolving as I can see that Duncan's test case suggestions have been implemented and they're passing.